syntax = "proto3";

package chat;

service ChatService {
    // stream bidirecional, permite enviar e receber mensagens entre duas pessoas
    rpc ChatStream (Empty) returns (stream ChatMessage);
    rpc SendMessage (ChatMessage) returns (Empty);
}

message Empty {}

message ChatMessage {
    string name = 1;
    string message = 2;
}

